query.go

```diff
@@ -126,6 +128,9 @@ type searchMessageResponse struct {
 func (c *Client) Search(request SearchRequest) ([]*Message, error) {
 	var buf strings.Builder
 
+	if request.Query == "" && request.Filters == nil && request.MessageFilters == nil {
+		return nil, errors.New("either query or filters must be provided")
```

Comment:
if `MessageFilters` isn't nil, it will skip this part. Seems behavior is different than error message. Is it intended?

Reply:
This is the idea. Filters (Channel filters) and message filters are filters). Error text is correct.

Reply:
OK, but it might not be obvious to the user from the error message because filters means both. How about

> one of the query, filters (channel) or message filters must be provided

Reply:
So we allow no filters, but deny nil? Note that you can call `len(request.Filters) != 0` on a nil Filters.

Reply:
Yes, this is very error-prone and length check is a very good idea.

However a *nit*; I think checking `len(someSlice) > 0` is superior to `len(someSlice) != 0`

because a length can be zero or bigger but not equal to zero somehow signals negative is possible.